java-checks/src/main/java/org/sonar/java/symexec/SymbolicEvaluator.java

```diff
@@ -370,177 +363,226 @@ public final void visitMemberSelectExpression(MemberSelectExpressionTree tree) {
     }
 
     @Override
-    void evaluateConditionalAnd(BinaryExpressionTree tree) {
-      PackedStates leftStates = evaluateCondition(currentState, tree.leftOperand());
-      currentResult = leftStates.getBooleanConstraint();
-      if (currentResult != SymbolicBooleanConstraint.FALSE) {
-        currentResult = null;
-        for (ExecutionState state : leftStates.trueStates) {
-          currentResult = evaluateExpression(state, tree.rightOperand()).union(currentResult);
-        }
-        if (currentResult != SymbolicBooleanConstraint.FALSE) {
-          currentResult = leftStates.getBooleanConstraint().union(currentResult);
+    public final void visitNewClass(NewClassTree tree) {
+      scan(tree.arguments());
+      SymbolicValue value = new SymbolicValue();
+      for (ExecutionState executionState : currentStates) {
+        SymbolicValue[] arguments = new SymbolicValue[tree.arguments().size()];
+        for (int i = arguments.length - 1; i >= 0; i -= 1) {
+          arguments[i] = executionState.pop();
         }
+        executionState.push(value);
+        check.onExecutableElementInvocation(executionState, tree, Arrays.asList(arguments));
       }
-      currentState.mergeRelations(Iterables.concat(leftStates.falseStates, leftStates.trueStates));
     }
 
     @Override
-    void evaluateConditionalOr(BinaryExpressionTree tree) {
-      PackedStates leftStates = evaluateCondition(currentState, tree.leftOperand());
-      currentResult = leftStates.getBooleanConstraint();
-      if (currentResult != SymbolicBooleanConstraint.TRUE) {
-        currentResult = null;
-        for (ExecutionState state : leftStates.falseStates) {
-          currentResult = evaluateExpression(state, tree.rightOperand()).union(currentResult);
+    public final void visitUnaryExpression(UnaryExpressionTree tree) {
+      super.visitUnaryExpression(tree);
+      if (tree.is(Tree.Kind.LOGICAL_COMPLEMENT)) {
+        for (ExecutionState executionState : currentStates) {
+          SymbolicBooleanConstraint constraint = executionState.getBooleanConstraint(executionState.peek());
+          if (constraint == SymbolicBooleanConstraint.FALSE) {
+            executionState.pop();
+            executionState.push(SymbolicValue.BOOLEAN_TRUE);
+          } else if (constraint == SymbolicBooleanConstraint.TRUE) {
+            executionState.pop();
+            executionState.push(SymbolicValue.BOOLEAN_FALSE);
+          }
         }
-        if (currentResult != SymbolicBooleanConstraint.TRUE) {
-          currentResult = leftStates.getBooleanConstraint().union(currentResult);
+      } else if (tree.is(Tree.Kind.POSTFIX_DECREMENT, Tree.Kind.POSTFIX_INCREMENT, Tree.Kind.PREFIX_DECREMENT, Tree.Kind.PREFIX_INCREMENT)) {
+        for (ExecutionState executionState : currentStates) {
+          Symbol.VariableSymbol symbol = extractVariableSymbol(tree.expression());
+          if (symbol != null) {
+            executionState.assignValue(symbol, new SymbolicValue());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this instantiation of "SymbolicValue". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2440)